multistep/commonsteps/step_http_server.go

```diff
@@ -22,23 +23,67 @@ import (
 //   http_port int - The port the HTTP server started on.
 type StepHTTPServer struct {
 	HTTPDir     string
+	HTTPContent map[string]string
 	HTTPPortMin int
 	HTTPPortMax int
 	HTTPAddress string
 
 	l *net.Listener
 }
 
+func StepHTTPServerFromHTTPConfig(cfg *HTTPConfig) *StepHTTPServer {
+	return &StepHTTPServer{
+		HTTPDir:     cfg.HTTPDir,
+		HTTPContent: cfg.HTTPContent,
+		HTTPPortMin: cfg.HTTPPortMin,
+		HTTPPortMax: cfg.HTTPPortMax,
+		HTTPAddress: cfg.HTTPAddress,
+	}
+}
+
+type MapServer map[string]string
+
+func (s MapServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	content, found := s[r.URL.Path]
+	if !found {
+		// TODO: this will be displayed on stdout, here we could implement a
+		// "did you mean" for helps.
+		http.Error(w, fmt.Sprintf("File %s not found", r.URL.Path), http.StatusNotFound)
+		return
+	}
+
+	if _, err := w.Write([]byte(content)); err != nil {
+		// log err in case the file couldn't be 100% transferred for example.
+		log.Printf("http_content serve error: %v", err)
+	}
+}
+
+func (s *StepHTTPServer) Handler() http.Handler {
```

Comment:
This one up close to the StepHTTPServer helps understand it faster.

Reply:
good one !